src/cdk/table/table.ts

```diff
@@ -357,11 +417,24 @@ export class CdkTable<T> implements CollectionViewer, OnInit, AfterContentChecke
 
   /** Set up a subscription for the data provided by the data source. */
   private _observeRenderChanges() {
-    this._renderChangeSubscription = this.dataSource.connect(this).pipe(takeUntil(this._onDestroy))
-      .subscribe(data => {
-        this._data = data;
-        this._renderRowChanges();
-      });
+    let dataStream: Observable<T[]> | undefined;
+
+    if (this.dataSource instanceof DataSource) {
```

Comment:
Weren't we going to check the properties instead of using `instanceof`? If people had been copying what `MatTableDataSource` was doing this would be a problem.

Reply:
I think we decided against it due to the possibility of property renaming/minification

Reply:
Ah, right.